src/commands/node/install.ts

```diff
@@ -0,0 +1,174 @@
+	networkSupported,
+	osSupported,
+} from '../../utils/node/utils';
+
+interface Flags {
+	readonly installationPath: string;
+	readonly name: string;
+	readonly network: NETWORK;
+	readonly 'no-snapshot': boolean;
+	readonly releaseUrl: string;
+	readonly snapshotUrl: string;
+}
+
+interface Options {
+	readonly installDir: string;
+	readonly liskTarSHA256Url: string;
+	readonly liskTarUrl: string;
+	readonly version: string;
+}
+
+const prereqCheck = (installPath: string): void => {
+	if (!osSupported()) {
+		throw new Error(`Lisk install is not supported on ${os.type()}`);
+	}
+	if (directoryExists(installPath)) {
+		throw new Error(`Installation already exists in path ${installPath}`);
+	}
+};
+
+const parseOptions = ({ network, releaseUrl, snapshotUrl }: Flags): void => {
+	networkSupported(network);
+	isValidURL(releaseUrl);
+	isValidURL(snapshotUrl);
+};
+
+const buildOptions = async ({
+	installationPath,
+	name,
+	network,
+	releaseUrl,
+}: Flags): Promise<Options> => {
+	const installPath = LISK_INSTALL(installationPath);
+	const installDir = `${installPath}/${name}/`;
+	const latestURL = LISK_LATEST_URL(releaseUrl, network);
+	const { version, liskTarUrl, liskTarSHA256Url } = await getReleaseInfo(
+		latestURL,
+		releaseUrl,
+		network,
+	);
+
+	return {
+		installDir,
+		version,
+		liskTarUrl,
+		liskTarSHA256Url,
+	};
+};
+
+const installLisk = async (options: Flags, cacheDir: string): Promise<void> => {
+	try {
+		const {
+			installDir,
+			liskTarUrl,
+			liskTarSHA256Url,
+			version,
+		} = await buildOptions(options);
+		const { name, network, snapshotUrl, 'no-snapshot': noSnapshot } = options;
+		const LISK_RELEASE_PATH = `${cacheDir}/${LISK_TAR(version)}`;
+		const LISK_RELEASE_SHA256_PATH = `${cacheDir}/${LISK_TAR_SHA256(version)}`;
+
+		prereqCheck(installDir);
+
+		await download(liskTarUrl, LISK_RELEASE_PATH);
+		await download(liskTarSHA256Url, LISK_RELEASE_SHA256_PATH);
+		await verifyChecksum(cacheDir, LISK_TAR_SHA256(version));
+
+		if (!noSnapshot) {
+			const snapshotPath = `${cacheDir}/${LISK_DB_SNAPSHOT(name, network)}`;
+			const snapshotURL = LISK_SNAPSHOT_URL(snapshotUrl, network);
+			await download(snapshotURL, snapshotPath);
+		}
+
+		createDirectory(installDir);
+		await extract(cacheDir, LISK_TAR(version), installDir);
+	} catch (error) {
+		throw new Error(`Lisk installation failed with error: ${error}`);
+	}
+};
+
+export default class InstallCommand extends BaseCommand {
+	static description = `Install lisk software`;
+
+	static examples = [
+		'node:install',
+		'node:install --installation-path=/opt/lisk/lisk-testnet --network=testnet',
+		'node:install --snapshot-url=no',
+	];
+
+	static flags = {
+		...BaseCommand.flags,
+		network: flagParser.string({
+			char: 'n',
+			description: 'Name of the network to install(mainnet, testnet, betanet).',
+			default: 'mainnet',
+		}),
+		installationPath: flagParser.string({
+			char: 'p',
+			description: 'Path of lisk software installation.',
```

Comment:
`Path of Lisk Core to install.`

Reply:
sure? I thought Lisk core was inappropriate, can we name it better and not use `core` anywhere?